packages/components/pie-cookie-banner/locales/index.js

```diff
@@ -0,0 +1,8 @@
+import enGB from './en-gb.json';
+import esES from './es-es.json';
+
+// TODO: Find how to tell TS each of these is of type CookieBannerLocale
+export default {
+    enGB,
+    esES,
+};
```

Comment:
Does anyone know how to make TS set the typing of these objects?

Reply:
Hey, I'm not an expert and I asked ChatGPT to help me 😂 

But something like this could perhaps work:

```ts
import * as enGBRaw from './en-gb.json';
import * as esESRaw from './es-es.json';

const enGB: CookieBannerLocale = enGBRaw;
const esES: CookieBannerLocale = esESRaw;

export default {
    enGB,
    esES,
};
```

It might require this setting in the TSconfig:
```json
{
  "compilerOptions": {
    "resolveJsonModule": true,
    "esModuleInterop": true,
    // ... other options
  }
}
```

Then we'd need to type the repeated parts of the JSON structure like this:

```ts
type LinkLabel = {
  label?: string;
  ariaLabel?: string;
};

type CookieCategory = {
  title: string;
  description?: string;
  cta?: LinkLabel;
};

type PreferencesManagement = {
  title: string;
  description: string;
  all: CookieCategory;
  necessary: CookieCategory;
  functional: CookieCategory;
  analytical: CookieCategory;
  personalized: CookieCategory;
  cta: {
    save: LinkLabel;
  };
};

type CookieBannerLocale = {
  banner: {
    title: string;
    description: string;
    cta: {
      managePreferences: string;
      necessaryOnly: string;
      acceptAll: string;
    };
  };
  preferencesManagement: PreferencesManagement;
};
```

I hope this might be of some use!

Reply:
That's appreciated, @jamieomaguire .

I was hoping a different approach could be used as this one can become a rabbit hole...

For experimentation sake, I updated `vite.config`, so it could simply build the locales TS:

<img width="1440" alt="image" src="https://github.com/justeattakeaway/pie/assets/448801/a39ea190-9284-4687-bdfa-46490110c010">

However it already fails on build:

<img width="1439" alt="image" src="https://github.com/justeattakeaway/pie/assets/448801/875c8cb9-39d0-4991-afff-2ec74d0c3906">

It keeps the English language in its own file, but somehow thinks the Spanish language should be embedded into `locales.js` 😵‍💫 .

<img width="1440" alt="image" src="https://github.com/justeattakeaway/pie/assets/448801/3c26a608-6f87-4a6e-ba7f-9db002347757">